staging/src/k8s.io/apiserver/pkg/endpoints/handlers/responsewriters/errors.go

```diff
@@ -76,3 +76,10 @@ func InternalError(w http.ResponseWriter, req *http.Request, err error) {
 		http.StatusInternalServerError)
 	utilruntime.HandleError(err)
 }
+
+// ServiceUnavailabeError renders a simple internal error
+func ServiceUnavailabeError(w http.ResponseWriter, req *http.Request, err error) {
+	http.Error(w, sanitizer.Replace(fmt.Sprintf("Service Unavailable: %q: %v", req.RequestURI, err)),
```

Comment:
Why an error page instead of a Status formatted JSON response?

Also did we debate the status code? e.g., what about 429 try again later? Can we at least add text that explains the error beyond "Service Unavailable"?

Reply:
I thought we had some error translation code somewhere already, maybe that's only for etcd sourced errors though?

Reply:
> Why an error page instead of a Status formatted JSON response?

This was probably copied from the other helper. Other filters we have (authorization, requestinfo, audit) write an error page when internal error occurs. I changed it to return a Status since we claim "[Kubernetes will always return the Status kind from any API endpoint when an error occurs](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#response-status-kind)". 

> the status code

The conversation happened in https://github.com/kubernetes/enhancements/pull/1361#discussion_r345463628. [429 Too Many Requests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#error-codes) sounds less suited IMO:
>> 429 StatusTooManyRequests 
Indicates that the either the client rate limit has been exceeded or the server has received more requests then it can process.
Suggested client recovery behavior:
Read the Retry-After HTTP header from the response, and wait at least that long before retrying.


> error translation code

perhaps this? https://github.com/kubernetes/kubernetes/blob/8bdd10b7d73cb0d705b895e0c9a5646dfbb19a24/staging/src/k8s.io/apiserver/pkg/endpoints/handlers/responsewriters/writers.go#L259